agent/agent.go

```diff
@@ -47,7 +47,10 @@ func StartAgent(c *cli.Context) {
 		d.Decode(msg)
 
 		// For now... will need to add some throttling at some point.
-		go handler.Handle(msg)
+		err := handler.Handle(msg)
+		if err != nil {
```

Comment:
Based on my usage, without the `go` it becomes a serialized event processor.

maybe we want:

```
func wrapHandler(..) {
   if err := f(msg); err != nil {
   logrus.error...
   }
} 

then:
go wrapHander(msg, handler.Handle)
```